### ISSUE TITLE
Vmware vAPP Parameters not getting applied when deploying Instance from Template

### ISSUE THREAD

**Issue by farhad-aslam** [apache/cloudstack#7354]
##### ISSUE TYPE

 * Bug Report

##### COMPONENT NAME

UI

##### CLOUDSTACK VERSION

4.17.2.0

##### CONFIGURATION

CS 4.17 with VMware vSphere 7.0.2

##### OS / ENVIRONMENT

N/A

##### SUMMARY

~~~
I am deploying an OEL VM using OVA template. The Template has vApp properties set on it but after deployment of VM, the parameter values are not set and VM fails to start giving error that Property 'IP_Address' must be configured for the VM to power on. In Vcenter I can see that VM is created by vApp Values are not set for any of the parameters. I have tested with multiple templates but getting same issue.
Same Template is working with below Scenarios:
1. Template is working fine with another Cloudstack deployment with CS 4.16
2. OVA is working fine when deploying VM directly from vSphere UI.
~~~
##### STEPS TO REPRODUCE

![image](https://user-images.githubusercontent.com/128424782/226468537-3d3306a0-cbd2-486f-b4db-f23eeb939e9b.png)

##### EXPECTED RESULTS

~~~
The VM was expected to have vAPP properties set
~~~

##### ACTUAL RESULTS

~~~
No parameter was set for the VM and VM failed to start.
~~~
![image](https://user-images.githubusercontent.com/128424782/226469278-88f03daa-01bc-4886-82a6-c682184be29c.png)

**Comment by nvazquez** [comment]
Hi @farhad-aslam thanks for reporting the issue. Is it possible to share the template or the .ovf file within the .ova file?

**Comment by farhad-aslam** [comment]
Hi @nvazquez please find attached the .ovf file from ova.
Please note that the same OVA is working fine with another deployment with 4.16 and also working fine while deploying directly on ESXi/vCenter.
I also tried by upgrading the problematic environment from 4.17 to 4.18 but the issues persists where VM gets created but fails to start because ovf properties are not set. It appears that the CS is not passing the parameters to vMware.
[Template.ovf.txt](https://github.com/apache/cloudstack/files/11064632/Template.ovf.txt)

**Comment by farhad-aslam** [comment]
Hi @nvazquez did you get a chance to have a look at this issue?
Meanwhile, I downgraded the application to 4.16 by restoring the DB backup and Management VM snapshots taken before upgrade, and ova issue did not appear on 4.16.

**Comment by nvazquez** [comment]
Thanks @farhad-aslam, sorry not updating you here, will be looking for a fix for 4.18.1

**Comment by JoaoJandre** [comment]
Hey @nvazquez are you still working on this?

**Comment by nvazquez** [comment]
Hi @JoaoJandre @farhad-aslam - I have worked on a similar issue which I suspect in can fix this problem. @farhad-aslam can you please test this fix? https://github.com/apache/cloudstack/pull/8072 (it will be available on 4.18.2)

**Comment by DaanHoogland** [comment]
a possible [fix](https://github.com/apache/cloudstack/pull/8072) has been mentioned and no interaction on this issue for a long while. I'll be closing this. Please fel free to add more info and re-open if it turns out to still be an issue.
